### PR TITLE
Add guide on spot with on-demand fall-back

### PR DESCRIPTION
**Pull request by marians**
Closes https://github.com/giantswarm/giantswarm/issues/11896

### CODE REVIEW

File: src/content/guides/spot-instances-with-on-demand-fallback/index.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+description: When using spot instances in a node pool on AWS, it can happen that the node pool cannot be scaled up as not enough spot instances are available. This guide shows you how to configure cluster-autoscaler in a way to provide on-demand instances as a back-up automatically.
+date: "2020-07-01"
+type: page
+weight: 120
+tags: ["tutorial"]
+---
+
+# Using on-demand instances as fall-back when spot instances are unavailable
+
+As of release v{{% first_aws_spotinstances_version %}} on AWS, node pools can use [spot instances](/basics/nodepools/#on-demand-spot). Users can select for each node pool which percentage of nodes should be covered by spot instances and by on-demand instances respectively. And users can define a base amount of on-demand instances that will be used, to ensure at least a certain amount of worker nodes is available at any time.
```
Comment (jw-s) [inline src/content/guides/spot-instances-with-on-demand-fallback/index.md:12]: to ensure at least a certain amount of worker nodes are available at any time.

Comment (marians) [inline src/content/guides/spot-instances-with-on-demand-fallback/index.md:12]:
```suggestion
As of release v{{% first_aws_spotinstances_version %}} on AWS, node pools can use [spot instances](/basics/nodepools/#on-demand-spot). Users can select for each node pool which percentage of nodes should be covered by spot instances and by on-demand instances respectively. And users can define a base amount of on-demand instances that will be used, to ensure at least a certain amount of worker nodes are available at any time.
```

File: src/content/guides/spot-instances-with-on-demand-fallback/index.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+This would try to create 10 pods, where each one would request 1 CPU and 1 GiB of memory.
+
+Once you apply this to the tenant cluster, watch the autoscaler logs using this command:
+
+```nohighlight
+kubectl -n kube-system logs -l app=cluster-autoscaler -f
+```
+
+The logs should show that cluster-autoscaler is registering the two Auto scaling Groups (ASGs) belonging to our node pools, similar to this:
```
Comment (marians) [inline src/content/guides/spot-instances-with-on-demand-fallback/index.md:151]:
```suggestion
The logs should show that cluster-autoscaler is registering the two Auto Scaling Groups (ASGs) belonging to our node pools, similar to this:
```

File: src/content/guides/spot-instances-with-on-demand-fallback/index.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+```nohighlight
+kubectl -n kube-system logs -l app=cluster-autoscaler -f
+```
+
+The logs should show that cluster-autoscaler is registering the two Auto Scaling Groups (ASGs) belonging to our node pools, similar to this:
+
+```nohighlight
+I0701 11:10:16.028372       1 auto_scaling_groups.go:138] Registering ASG cluster-cp983-tcnp-sp0ti-NodePoolAutoScalingGroup-139WETDOF86WA
+I0701 11:10:16.028394       1 auto_scaling_groups.go:138] Registering ASG cluster-cp983-tcnp-0ndmd-NodePoolAutoScalingGroup-1B8IU42LTSOYU
```
Comment (marians) [inline src/content/guides/spot-instances-with-on-demand-fallback/index.md:155]:
```suggestion
I0701 11:10:16.028372       1 auto_scaling_groups.go:138] Registering ASG cluster-abc12-tcnp-sp0ti-NodePoolAutoScalingGroup-139WETDOF86WA
I0701 11:10:16.028394       1 auto_scaling_groups.go:138] Registering ASG cluster-abc12-tcnp-0ndmd-NodePoolAutoScalingGroup-1B8IU42LTSOYU
```

File: src/content/guides/spot-instances-with-on-demand-fallback/index.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+```nohighlight
+I0701 11:10:16.028372       1 auto_scaling_groups.go:138] Registering ASG cluster-abc12-tcnp-sp0ti-NodePoolAutoScalingGroup-139WETDOF86WA
+I0701 11:10:16.028394       1 auto_scaling_groups.go:138] Registering ASG cluster-abc12-tcnp-0ndmd-NodePoolAutoScalingGroup-1B8IU42LTSOYU
+```
+
+When the deployment has been scaled up, some line like this should appear:
+
+```nohighlight
+I0701 11:16:47.077234       1 priority.go:167] priority expander: cluster-cp983-tcnp-sp0ti-NodePoolAutoScalingGroup-139WETDOF86WA chosen as the highest available
```
Comment (marians) [inline src/content/guides/spot-instances-with-on-demand-fallback/index.md:161]:
```suggestion
I0701 11:16:47.077234       1 priority.go:167] priority expander: cluster-abc12-tcnp-sp0ti-NodePoolAutoScalingGroup-139WETDOF86WA chosen as the highest available
```

File: src/content/guides/spot-instances-with-on-demand-fallback/index.md (hunk 1 above)
Comment (marians) [reply src/content/guides/spot-instances-with-on-demand-fallback/index.md]: Modified, thanks

**Pull request merged**